src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMissingWhitespaceAfterAsteriskCheck.java

```diff
@@ -0,0 +1,141 @@
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks that there is at least one whitespace after the leading asterisk.
+ * Although spaces after asterisks are optional in the Javadoc comments, their absence
+ * makes the documentation difficult to read. It is the de facto standard to put at least
+ * one whitespace after the leading asterisk.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">Tight-HTML Rules</a>.
+ * Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the default check:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocMissingWhitespaceAfterAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Code Example:
+ * </p>
+ * <pre>
+ * &#47;** This is valid single-line Javadoc. *&#47;
+ * class TestClass {
+ *   &#47;**
+ *     *This is invalid Javadoc.
+ *     *&#47;
+ *   int invalidJavaDoc;
+ *   &#47;**
+ *     * This is valid Javadoc.
+ *     *&#47;
+ *   void validJavaDocMethod() {
+ *   }
+ *   &#47;**This is invalid single-line Javadoc. *&#47;
+ *   void invalidSingleLineJavaDocMethod() {
+ *   }
+ *   &#47;** This is valid single-line Javadoc. *&#47;
+ *   void validSingleLineJavaDocMethod() {
+ *   }
+ * }
+ * </pre>
+ *
+ * @since 8.32
+ */
+@StatelessCheck
+public class JavadocMissingWhitespaceAfterAsteriskCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_KEY = "javadoc.missing.whitespace";
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+            JavadocTokenTypes.LEADING_ASTERISK,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode detailNode) {
+        final DetailNode textNode;
+
+        if (detailNode.getType() == JavadocTokenTypes.JAVADOC) {
+            textNode = JavadocUtil.getFirstChild(detailNode);
+        }
+        else {
+            textNode = JavadocUtil.getNextSibling(detailNode);
+        }
+
+        if (textNode != null
+                && textNode.getType() != JavadocTokenTypes.EOF
+                && !hasWhitespaceAfterAsteriskBeforeText(textNode)) {
+            log(textNode.getLineNumber(), textNode.getColumnNumber(), MSG_KEY);
```

Comment:
unfortunately I still do not like design .....
name `hasWhitespaceAfterAsteriskBeforeText` is not same as logic inside, it will be a problems in future.

please do:
```
if (textNode != null
  && textNode.getType() != JavadocTokenTypes.EOF) {
    int lastAsterisPosition = getLastLeadingAsterik(....)
    if (!isLast(lastAsterisPosition) && !isWhitespace(lastAsterisPosition +1  )) {
      log(textNode.getLineNumber(), textNode.getColumnNumber(), MSG_KEY);
    }
}
```

and please add tests case with tabs after leading asterik.
Please create new Input file for this as tabs are forbidden in java files in our repo, so we need to suppress violation on it.

Reply:
done

Reply:
> Please create new Input file for this as tabs are forbidden in java files in our repo, so we need to suppress violation on it.

sad leftover #8019, we do not validate such code for bad whitespaces.